src/NuGet.Clients/NuGet.PackageManagement.UI/Models/NuGetProjectUpgradeWindowModel.cs

```diff
@@ -0,0 +1,231 @@
+                {
+                    InitPackageUpgradeIssues();
+                }
+                return _errors;
+            }
+        }
+
+        public IList<PackageIdentity> NotFoundPackages
+        {
+            get
+            {
+                if (_notFoundPackages == null)
+                {
+                    InitPackageUpgradeIssues();
+                }
+                return _notFoundPackages;
+            }
+        }
+
+        public bool HasIssues => Errors.Any() || Warnings.Any();
+
+        public IEnumerable<NuGetProjectUpgradeDependencyItem> UpgradeDependencyItems
+            => _upgradeDependencyItems ?? (_upgradeDependencyItems = GetUpgradeDependencyItems());
+
+        public IEnumerable<string> DirectDependencies => CollapseDependencies ? IncludedCollapsedPackages : AllPackages;
+
+        public IEnumerable<string> TransiviteDependencies => CollapseDependencies ? DependencyPackages : new List<string>();
+
+        private IEnumerable<string> DependencyPackages => _dependencyPackages ?? (_dependencyPackages = GetDependencyPackages());
+
+        private IEnumerable<string> AllPackages => _allPackages ?? (_allPackages = UpgradeDependencyItems.Select(d => d.Package.ToString()));
+
+        private IEnumerable<string> IncludedCollapsedPackages => _includedCollapsedPackages ?? (_includedCollapsedPackages = GetIncludedCollapsedPackages());
+
+        private IEnumerable<string> GetDependencyPackages()
+        {
+            return UpgradeDependencyItems.Where(d => d.DependingPackages.Any()).Select(d => d.ToString());
+        }
+
+        private IEnumerable<string> GetIncludedCollapsedPackages()
+        {
+            return UpgradeDependencyItems
+                .Where(upgradeDependencyItem => !upgradeDependencyItem.DependingPackages.Any())
+                .Select(upgradeDependencyItem => upgradeDependencyItem.Package.ToString());
+        }
+
+        private void InitPackageUpgradeIssues()
+        {
+            _warnings = new List<string>();
+            _errors = new List<string>();
+            _notFoundPackages = new List<PackageIdentity>();
+
+            var msBuildNuGetProject = (MSBuildNuGetProject)Project;
+            var framework = msBuildNuGetProject.ProjectSystem.TargetFramework;
+            var folderNuGetProject = msBuildNuGetProject.FolderNuGetProject;
+
+            foreach (var package in PackageDependencyInfos)
+            {
+                // We create a new PackageIdentity here, otherwise we would be passing in a PackageDependencyInfo
+                // which includes dependencies in its ToString().
+                InitPackageUpgradeIssues(folderNuGetProject, new PackageIdentity(package.Id, package.Version), framework);
+            }
+        }
+
+        private void InitPackageUpgradeIssues(FolderNuGetProject folderNuGetProject, PackageIdentity packageIdentity, NuGetFramework framework)
+        {
+            // Confirm package exists
```

Comment:
is `lib/` checked?

Reply:
the validation will be re-worked in a subsequent PR.